libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/MapboxNavigationAccounts.kt

```diff
@@ -36,27 +36,32 @@ class MapboxNavigationAccounts private constructor() : SkuTokenProvider {
             skuGenerator = when (Billing.getInstance(context).getBillingType()) {
                 Billing.BillingModel.MAU -> MauSku(
                     preferences,
-                    TIMER_EXPIRE_AFTER * MAU_TIMER_EXPIRE_THRESHOLD
-                )
-                Billing.BillingModel.TRIPS -> TripsSku(
-                    preferences, TIMER_EXPIRE_AFTER * TRIPS_TIMER_EXPIRE_THRESHOLD,
-                    TRIPS_REQUEST_COUNT_THRESHOLD
+                    TIMER_EXPIRE_AFTER * MAU_TIMER_EXPIRE_THRESHOLD,
+                    context.applicationContext.packageName
                 )
+                Billing.BillingModel.TRIPS -> null
             }
         }
     }
 
-    override fun obtainSkuToken(resourceUrl: String?, querySize: Int): String {
-        val skuToken = skuGenerator?.generateToken() ?: ""
+    @Synchronized
```

Comment:
Is `obtainUrlWithSkuToken` accessed from different threads? Want to double check if `@Synchronized` is really needed.

Reply:
Yes I would like this to be here, because this is gonna end up being called from various places like `faster route`, `off route`, `route refresh`. Situations like faster route has a timer which fires every 2 minutes and there might be race conditions when route refresh or faster route trigger at the same time. In that situation we need it to be `synchronized`